src/inline-loading/inline-loading.component.ts

```diff
@@ -14,60 +27,88 @@ import {
 @Component({
 	selector: "ibm-inline-loading",
 	template: `
-		<div class="bx--inline-loading__animation">
+		<div *ngIf="state !== InlineLoadingState.Hidden"
+			class="bx--inline-loading__animation">
 			<div
-				*ngIf="success === false"
+				*ngIf="state === InlineLoadingState.Inactive || state === InlineLoadingState.Active"
 				class="bx--loading bx--loading--small"
 				[ngClass]="{
-					'bx--loading--stop': !isActive
+					'bx--loading--stop': state === InlineLoadingState.Inactive
 				}">
 				<svg class="bx--loading__svg" viewBox="-75 -75 150 150">
 					<circle class="bx--loading__background" cx="0" cy="0" r="30" />
 					<circle class="bx--loading__stroke" cx="0" cy="0" r="30" />
 				</svg>
 			</div>
 			<svg
-				*ngIf="success === true"
+				*ngIf="state === InlineLoadingState.Finished"
 				class="bx--inline-loading__checkmark-container bx--inline-loading__svg"
 				xmlns="http://www.w3.org/2000/svg"
 				viewBox="0 0 10 10">
 				<polyline class="bx--inline-loading__checkmark" points="0.74 3.4 3.67 6.34 9.24 0.74"></polyline>
 			</svg>
```

Comment:
while we're here we should probably change this to use the directive as well, like the one in https://github.com/IBM/carbon-components-angular/blob/master/src/notification/notification.component.ts#L39-L44

Reply:
Certainly :)
No code duplication is allowed! :)